packages/expo-splash-screen/src/index.native.ts

```diff
@@ -4,80 +4,56 @@ import { SplashScreenNativeModule, SplashScreenOptions } from './SplashScreen.ty
 
 const SplashModule = requireOptionalNativeModule<SplashScreenNativeModule>('ExpoSplashScreen');
 
-let _userControlledAutoHideEnabled = false;
-let _preventAutoHideAsyncInvoked = false;
-
-/**
- * Expo Router uses this internal method to ensure that we can detect if the user
- * has explicitly opted into preventing the splash screen from hiding. This means
- * they will also explicitly hide it. If they don't, we will hide it for them after
- * the navigation render completes.
- *
- * @private
- */
-export async function _internal_preventAutoHideAsync(): Promise<boolean> {
+export function setOptions(options: SplashScreenOptions) {
   if (!SplashModule) {
-    return false;
+    return;
   }
 
-  // Memoize, this should only be called once.
-  if (_preventAutoHideAsyncInvoked) {
-    return false;
-  }
-  _preventAutoHideAsyncInvoked = true;
+  SplashModule.setOptions(options);
+}
 
-  // Append error handling to ensure any uncaught exceptions result in the splash screen being hidden.
-  // This prevents the splash screen from floating over error screens.
-  if (ErrorUtils?.getGlobalHandler) {
-    const originalHandler = ErrorUtils.getGlobalHandler();
-    ErrorUtils.setGlobalHandler((error, isFatal) => {
-      hide();
-      originalHandler(error, isFatal);
-    });
+export function hide() {
+  if (!SplashModule) {
+    return;
   }
```

Comment:
this block is the only thing that i didn't port over. cc @alanjhughes - this seems like something we would want to build into expo-splash-screen, perhaps on the native side, if we don't already handle this in the case of errors. although i believe that now that we use the loading view approach it may not be necessary

Reply:
We should be able to do this from native but probably safer to keep this until we verify for sure it's no longer needed

Reply:
We should be able to do this from native but probably safer to leave this here for now until we verify it's not needed

Reply:
done